src/main/java/org/terasology/climateConditions/HypothermiaSystem.java

```diff
@@ -59,4 +97,24 @@ public void modifySpeed(GetMaxSpeedEvent event, EntityRef player) {
     public void modifyJumpSpeed(AffectJumpForceEvent event, EntityRef player) {
         event.multiply(jumpSpeedMultiplier);
     }
+
+    private void applyFrostbiteDamagePlayer(EntityRef player) {
+        Prefab frostbiteDamagePrefab = prefabManager.getPrefab("ClimateConditions:FrostbiteDamage");
+        player.send(new DoDamageEvent(healthDecreaseAmount, frostbiteDamagePrefab));
+    }
+
+    private void applyStunEffect(EntityRef player, int duration) {
+        StunAlterationEffect stunAlterationEffect = new StunAlterationEffect(context);
+        //Both the instigator and the target is the player
+        //the magnitude parameter is not used by StunAlterationEffect
+        stunAlterationEffect.applyEffect(player, player, 0, duration);
+    }
+
+    public void playFrostbiteSound(EntityRef entity) {
+        CharacterSoundComponent characterSounds = entity.getComponent(CharacterSoundComponent.class);
+        if (characterSounds.deathSounds.size() > 0) {
```

Comment:
If the player does not have a `CharacterSoundComponent` this will throw a NPE.
```suggestion
        if (characterSounds != null && characterSounds.deathSounds.size() > 0) {
```